### PR TITLE
Get scaling config and cron schedule from vars

### PR DESCRIPTION
**Pull request by ethanroday**
- Parameterize lambda module to provide scaling config based on variables in root config
- Add cron expressions for scaling at the right time of day
- Pass scaling config to lambda function through CW event rule input

NOTE: The cron configuration is currently set to upscale every 30 minutes and downscale every 30 minutes (offset by 15) for testing purposes only! Will change to correct config when going to staging/prod.

### CODE REVIEW

**Review by tiagojsag — DISMISSED** [review]
There are a bunch of things here that I'm not familiar with, so maybe my review is not very valuable, but it seems good to me :+1:

File: lambda/eks_scaling/src/eks_scaling.py [hunk 1]
```diff
@@ -41,28 +23,48 @@ def should_update_scaling_config(client, nodegroup_name, desired_scaling_config)
...
+    #       "maxSize": 123,
+    #       "desiredSize": 123
+    #     },
+    #     "gfw-node-group": {
+    #       "minSize": 123,
+    #       "maxSize": 123,
+    #       "desiredSize": 123
+    #     }
+    #   }
+    # }
```
Comment (thomas-maschler) [inline lambda/eks_scaling/src/eks_scaling.py]: I would slightly change the payload structure to make it more scalable:

```json
[{
	"eks_cluster_name": "cluster-name",
	"node_groups": [{
			"node_group_name": "apps-node-group",
			"scaling_config": {
				"minSize": 123,
				"maxSize": 123,
				"desiredSize": 123
			}
		},
		{
			"node_group_name": "gfw-node-group",
			"scaling_config": {
				"minSize": 123,
				"maxSize": 123,
				"desiredSize": 123
			}
		}
	]
}]
```

**Review by thomas-maschler — CHANGES_REQUESTED** [review]
AWS creates a logging group for the lambda function implicitly. 
You will have to specify it in terraform. 
Since AWS already created it for you, you will first have to delete the log group via the console before you deploy your new code. Otherwise there will be a conflict.

I would also rename the module name to something close to what is does ("cloud watch events" if you plan to add additional events here or "eks auto scaler" if you plan to only use it for this one feature)

```
resource "aws_cloudwatch_log_group" "lambda" {
  name              = "/aws/lambda/eks_scaling" // verify log group name
  retention_in_days = var.log_retention
}
```

File: lambda/eks_scaling/src/eks_scaling.py (hunk 1 above)
Comment (ethanroday) [reply lambda/eks_scaling/src/eks_scaling.py]: Sounds reasonable to me. Fixed.

**Comment by ethanroday** [comment]
Added log group explicitly and deleted the existing one in the console.

**Review by thomas-maschler — CHANGES_REQUESTED** [review]
Looks good.
One last thing. I assume the scaling logic will vary across environments, correct?
In that case I would not set default values for the variables, but instead set them in 

`vars/terraform-dev.tfvars`
`vars/terraform-staging.tfvars`
`vars/terraform-production.tfvars`

accordingly

**Comment by ethanroday** [comment]
Thanks for bringing that up - was actually going to ask how that would work when going to staging. I'll move the configs - just a minute.

**Comment by ethanroday** [comment]
Okay done. For now I've just put _all_ of the scaling config for `apps-node-group` and `gfw-node-group` into the `.tfvars` file, since we may want to vary the base configs by environment as well.

**Comment by ethanroday** [comment]
Oops...plan is failing now.

**Comment by thomas-maschler** [comment]
You still need to declare the variables in `variables.tf` just leave out the default value.
Set the value in the tfvar files

**Comment by ethanroday** [comment]
Fixed. I left the default value as the same as the downscaled value just in case they ever get deleted accidentally in tfvars.

**Comment by thomas-maschler** [comment]
It would then prompt you to provide a value during plan

**Review by thomas-maschler — APPROVED** [review]

**Pull request merged**